File: 1.py
m = 0

def getFuel(f):
    fu = f // 3 - 2
    if fu > 0:
        return fu + getFuel(fu)
    else:
        return 0

with open("1.txt") as F:
    for line in F:
        m = m + getFuel(int(line))
        
print(m)

File: aoc2021/1.rs
use std::path::Path;
use std::io::{self,BufRead};
use std::fs::File;

fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}


fn main(){
    let mut count = 0;
    if let Ok(lines) = read_lines("./1.txt"){
        let mut a = 10000;
        let mut b = 10000;
        let mut c = 10000;
        for line in lines{
            if let Ok(s) = line{
                let i = s.parse().unwrap();
                if i+c+b > a+b+c {
                    count = count+1;
                }
                a = b;
                b = c;
                c = i;
            }
        }
    }
    println!("{:?}",count)
}
